src/metadata_serializer.c

```diff
@@ -0,0 +1,119 @@
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <endian.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "metadata_serializer.h"
+
+struct sedcli_metadata *sedcli_metadata_alloc_buffer()
+{
+	uint8_t *buffer;
+
+	buffer = malloc(SEDCLI_METADATA_SIZE);
+	if (buffer == NULL) {
+		return NULL;
+	}
+
+	memset(buffer, 0, SEDCLI_METADATA_SIZE);
+
+	return (struct sedcli_metadata *)buffer;
+}
+
+void sedcli_metadata_free_buffer(struct sedcli_metadata *buffer)
+{
+	if (buffer != NULL) {
+		memset(buffer, 0, SEDCLI_METADATA_SIZE);
+		free(buffer);
+	}
+}
+
+void sedcli_metadata_init(struct sedcli_metadata *meta, uint32_t pek_id_size,
+		uint32_t iv_size, uint32_t enc_dek_size, uint32_t tag_size)
+{
+	meta->magic_num = htole64(SEDCLI_META_MAGIC);
+	meta->version = htole32(SEDCLI_META_VERSION);
+
+	meta->pek_id_size = htole32(pek_id_size);
+	meta->iv_size = htole32(iv_size);
+	meta->enc_dek_size = htole32(enc_dek_size);
+	meta->tag_size = htole32(tag_size);
+}
+
+uint8_t *sedcli_meta_get_pek_id_addr(struct sedcli_metadata *meta)
+{
+	return meta->data;
+}
+
+uint8_t *sedcli_meta_get_iv_addr(struct sedcli_metadata *meta)
```

Comment:
starting from this function, I would recommend checking the 'meta' pointer for NULL in all the future functions.